cpp/src/arrow/compute/kernels/scalar_nested.cc

```diff
@@ -269,14 +312,34 @@ struct StructFieldFunctor {
   }
 };
 
+Result<const DataType*> RecursiveResolveStructFieldType(const FieldRef& field_ref,
```

Comment:
This could maybe also be replaced or simplified with one of the existing FieldRef / FieldPath utilities?

Reply:
Good eye! :eyes: